packages/legend-query/src/stores/QueryBuilderExplorerState.ts

```diff
@@ -139,19 +169,47 @@ export const buildPropertyExpressionFromExplorerTreeNodeData = (
     multiplicityOne,
   );
   propertyExpression.func = guaranteeNonNullable(node.property);
-  let currentExpression = propertyExpression;
+  let currentExpression: AbstractPropertyExpression | SimpleFunctionExpression =
+    propertyExpression;
   let parentNode = treeData.nodes.get(node.parentId);
-  while (parentNode instanceof QueryBuilderExplorerTreePropertyNodeData) {
-    const parentPropertyExpression = new AbstractPropertyExpression(
-      '',
-      multiplicityOne,
-    );
-    parentPropertyExpression.func = guaranteeNonNullable(parentNode.property);
+  let currentNode: QueryBuilderExplorerTreeNodeData = node;
+  while (
+    parentNode instanceof QueryBuilderExplorerTreePropertyNodeData ||
+    parentNode instanceof QueryBuilderExplorerTreeSubTypeNodeData
+  ) {
+    let parentPropertyExpression;
+    if (parentNode instanceof QueryBuilderExplorerTreeSubTypeNodeData) {
+      parentPropertyExpression = new SimpleFunctionExpression(
+        'subType',
```

Comment:
Can we not hardcode this. Let's do it like other supported types. Example: `extractElementNameFromPath(SUPPORTED_FUNCTIONS.GET_ALL)`